src/main/java/com/zendesk/maxwell/MaxwellReplicator.java

```diff
@@ -79,6 +81,13 @@ private void ensureReplicatorThread() throws Exception {
 			LOGGER.warn("open-replicator stopped at position " + replicator.getBinlogFileName() + ":" + replicator.getBinlogPosition() + " -- restarting");
 			replicator.start();
 		}
+
+		Long ms = replicator.millisSinceLastEvent();
+		if ( ms != null && ms > 2000 ) {
+			LOGGER.warn("no heartbeat heard from server in " + ms + "ms.  restarting replication.");
+			replicator.stop(5, TimeUnit.SECONDS);
```

Comment:
no, that only applies if we need to shut down the whole system.  bouncing the replicator underneath the event listener is fine.